### PR TITLE
add line for Daily Latest MAE all GSPs to fig6

### LINKED ISSUE #29: Add metric sum of all gsp
## Detailed Description
Add metric on metric page that displays "Daily Latest MAE All GSPs". 

## Context
- This gives an metrix of how good the forecast is predicting for all the GSPs

## Possible Implementation
- load another metric using name="Daily Latest MAE All GSPs" and plot on graph. There will be one data point for each day

### PR DESCRIPTION
**Pull request by rachel-labri-tipton**
# Pull Request

## Description

This adds code to `main.py`, the `metric page`, to display `Daily Lates MAE All GSPs`. 
Currently the data frame I created for displaying `Daily Latest MAE All GSPs` returns an empty array, so I'd be curious to have answers to the following: 
- how best to query the database in cases like this to see if there is the data for `Daily Latest MAE All GSPs`
-  why the data frame I've added on `line 180` of `main.py`, `df_mae_all_gsp`, is empty

I tried the following in the pgAdmin database to see if the data's there:
`select * from metric_value 
where created_utc >='2023-06-28'
and metric_id = 2`
and did see values in the `value` column: 
Screenshot 2023-07-05 at 14 47 52

Theoretically the new line would be in the following graph, `fig6` in `main.py`:

Screenshot 2023-07-05 at 14 45 31

Fixes #29

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
Before merging this, lets make sure we see the plot

File: src/main.py [hunk 1]
```diff
@@ -111,11 +113,22 @@ def metric_page():
             end_datetime_utc=endtime,
             model_name=model_name,
         )
+        # get metric value for mae with pvlive gsp sum truths for comparison
+        metric_values_mae_gsp_sum = get_metric_value(
+            session=session,
+            name=name_mae_gsp_sum,
+            gsp_id=0,
```
Comment (peterdudfield) [inline src/main.py:120]: could you try with out this gsp_id=0, and i think th esum of trusths does not have a gsp_id

**Review by peterdudfield — CHANGES_REQUESTED** [review]
Lets get the data on the plot first

**Comment by rachel-labri-tipton** [comment]
I've gotten `Daily Latest MAE All GSPs` to plot when `model_name` is `cnn` or `pvnet_v2`. Otherwise there's no data for National_xg, which makes sense. 
Screenshot 2023-07-07 at 16 00 58

File: src/main.py [hunk 2]
```diff
@@ -380,8 +400,32 @@ def metric_page():
...
+            y=df_mae_all_gsp["MAE All GSPs"],
+            mode="lines",
+            name="Daily Latest MAE All GSPs",
+            line=dict(color=line_color[4]),
+        ),
+        )
+    
+    if model_name in ["pvnet_v2", "cnn"]:
+        
+        st.plotly_chart(fig6, theme="streamlit")
```
Comment (peterdudfield) [inline src/main.py:427]: how come fig6 is also in this?

**Review by peterdudfield — CHANGES_REQUESTED** [review]
one smal change/question, looks good otherwise.

Also you might need to solve a git conflict, but hopefully that'll b ok

File: src/main.py (hunk 2 above)
Comment (rachel-labri-tipton) [reply src/main.py:427]: so fig6 shows RMSE and there isn't any RMSE data when the `National_xg` model is selected. it just plots MAE, which is already displayed at top of the page. should I go ahead and leave the chart in?

Comment (peterdudfield) [reply src/main.py:427]: I think there shuold be a RMSE for `National_xg`, perhaps just it in for the moment, and we can sort it out if RMSE is not showing for `National_xg`

File: src/main.py [hunk 3]
```diff
@@ -402,6 +402,7 @@ def metric_page():
     )
 
     fig6.update_layout(yaxis_range=[0, MAE_LIMIT_DEFAULT])
+    st.plotly_chart(fig6, theme="streamlit")
```
Comment (rachel-labri-tipton) [inline src/main.py:405]: I added the fig6 MAE/RMSE chart back in for all data models.

Comment (peterdudfield) [reply src/main.py:405]: nice thanks

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**